File: sdks/ts/src/sdk/models/shared/enterprisebase.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum EnterpriseBaseTypeEnum {
    Enterprise = "enterprise"
}


// EnterpriseBase
/** 
 * A mini representation of a enterprise, used when
 * nested within another resource.
**/
export class EnterpriseBase extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: EnterpriseBaseTypeEnum;
}

File: sdks/go/pkg/models/shared/enterprisebase.go
package shared

type EnterpriseBaseTypeEnum string

const (
	EnterpriseBaseTypeEnumEnterprise EnterpriseBaseTypeEnum = "enterprise"
)

// EnterpriseBase
// A mini representation of a enterprise, used when
// nested within another resource.
type EnterpriseBase struct {
	ID   *string                 `json:"id,omitempty"`
	Type *EnterpriseBaseTypeEnum `json:"type,omitempty"`
}
